### ISSUE TITLE
useWebSocket : changge url heartbeat no work , when message: new TextEncoder().encode('{}')

### ISSUE THREAD

**Issue by a982246809** [vueuse/vueuse#3706]
### Describe the bug

 Below is the minimum reproduction
 下面是最小复现
 
 const url = ref('ws://127.1.1.1/?sid=')
 setTimeout(()=>url.value='ws://127.1.1.1/?sid=52617237-df6d-4ec2-8d53-c7ad91e25791',3000)
 useWebSocket(url, {
    autoReconnect:false,
     heartbeat: {
        message: new TextEncoder().encode('{}'),
      },

### Reproduction

https://stackblitz.com/edit/vitejs-vite-upu45d/?file=src%2FApp.vue

### System Info

```Shell
System:
    OS: Windows 11 10.0.22631
    CPU: (16) x64 12th Gen Intel(R) Core(TM) i7-12650H
    Memory: 8.05 GB / 31.73 GB
  Binaries:
    Node: 20.10.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.2.3 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.2.0 - ~\AppData\Local\pnpm\pnpm.CMD
  Browsers:
    Edge: Chromium (120.0.2210.121)
    Internet Explorer: 11.0.22621.1
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by a982246809** [comment]
![image](https://github.com/vueuse/vueuse/assets/56909676/9080f861-b240-4058-9188-f7a3e5f3fe97)
this is undefind ; 

wsRef.value.close(code, reason) does not seem to trigger ws.onclose immediately
wsRef.value.close(code, reason) 似乎不会立刻触发   ws.onclose

After the url change triggers open(), the previous ws.onclose is executed, resulting in wsRef.value==undefined
导致url改变触发open()后 , 上一个ws.onclose执行 , 导致wsRef.value==undefined

**Comment by ilyaliao** [comment]
Closed via #3707
